src/eval.c

```diff
@@ -728,6 +730,16 @@ NULL
             addReplyError(c, "Use SCRIPT DEBUG YES/SYNC/NO");
             return;
         }
+    } else if (c->argc == 3 && !strcasecmp(c->argv[1]->ptr, "dump")) {
+        dictEntry *de;
+        luaScript *ls;
+
+        if (sdslen(c->argv[2]->ptr) == 40 && (de = dictFind(lctx.lua_scripts, c->argv[2]->ptr))) {
```

Comment:
Here if you reference script exists command how to check the input parameter, you will see the c->argv[2]->ptr length does not been checked. So my opinion is that if you want to length of argv[2]->ptr, you had better prompt a much clearer information to the client, such as "sha1 length is not correct"

Reply:
This was discussed in an early revision, and I believe we ended on this. Unless you feel strongly I think we should leave it as is.

Reply:
I vote to leave it as is.

This is just a fast path to return error. Not a real validation. If we want a real validation, we should also check for invalid characters in the SHA1. Let's not do that.